### PR TITLE
Allowing easy-install plugin with bash script

### PR DESCRIPTION
**Pull request by propilideno**
# Description

### The goal of this pull request is:
- Allow the user to easy-install **OntoUML Plugin** for Visual Paradigm.
- Fix the issue: #145

# Checklist
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes are relevant to the project.

# WhoAmI
I am Propi (Lucas de Almeida), i'd contributed with Open Source Projects and with Master's degree articles. I also have advanced skills in bash scripting and my changes are huge relevant for this project.
### Feel free to contact me.
- My discord: Propi#3308
- My email: contato@propi.dev

# Changes made
### Reused from [FrameWeb-VP-Plugin Fork](https://github.com/propilideno/frameweb-vp-plugin)
- Verify the **default paths** to app and plugin and ask to confirm the typed path.
- Download the ontouml-vp-Repository and clean it after installation of plugin
- Verify **JDK** and **Maven** installation, if uninstalled the script is able to install this requirements in UNIX OS.
- In Windows, this script can open the **Tutorial** of Maven and JDK Installation.
- Can install shell requirements like **HomeBrew**
- Can modify `pom.xml` to **App dir** and **Plugin dir** given directories or default path.
- Easy-to-maintain with a lot of functions with good practices.
### My adaptations for OntoUML
- Updated the script to support OntoUML automatic installation.
- The code was refactored improving encapsulation from the `get_VP_App_Path` and `get_VP_Plugin_Path` functions.
- I put the curl with bash script under [User Instructions](https://github.com/propilideno/ontouml-vp-plugin#user-instructions), but you can adapt it like @vitorsouza did in [FrameWeb](https://github.com/nemo-ufes/frameweb-vp-plugin)
- Was commented `install_maven` because in OntoUML, maven is optional. So, we're installing with **Maven Wrapper**

# Testing
Open `Git Bash` or `Terminal`, and type:
```
bash <(curl -sL https://raw.githubusercontent.com/propilideno/ontouml-vp-plugin/master/install.sh)
```

# Final Thoughts
I refrained from including my name in the README file since I am not a member of the team. However, I would be honored if you would consider crediting me for my contributions.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+#App Default Path
+VISUAL_PARADIGM_APP_DIR_WINDOWS="C:\\Program Files\\Visual Paradigm CE 17.0\\" #Windows
+VISUAL_PARADIGM_APP_DIR_MAC="/Applications/Visual Paradigm.app/Contents/Resources/app/" #MacOS
+VISUAL_PARADIGM_APP_DIR_LINUX="/home/$USER/Visual_Paradigm_17.0/" #Linux
+#Plugin Default Path
+VISUAL_PARADIGM_PLUGIN_DIR_WINDOWS="C:\\Users\\$USER\\AppData\\Roaming\\VisualParadigm\\plugins\\" #Windows
+VISUAL_PARADIGM_PLUGIN_DIR_MAC="/Users/$USER/Library/Application Support/VisualParadigm/plugins/" #MacOS
+VISUAL_PARADIGM_PLUGIN_DIR_LINUX="/home/$USER/.config/VisualParadigm/plugins/" #Linux
+
+function read_App_Path(){
```
Comment (matheuslenke) [inline install.sh:21]: This function does not work on Windows when trying to change the default path of Visual Paradigm. Changing to this is working:

```bash
function read_App_Path(){
    vp_path="$1"
    while true; do
        echo "Visual Paradigm Path: $vp_path"
        read -p "Confirm (y/n)?: " choice
        case "$choice" in
            y|Y ) 
                if [[ -d "$vp_path" ]]; then
                    break
                else
                    printf "<FOLDER NOT FOUND> Type a valid path!\n"
                fi
            ;;
            n|N ) 
                echo "The path to your Visual Paradigm (APP FOLDER) is: "
                read -r vp_path ;;
            * ) printf "Invalid input\n";;
        esac
    done
    echo "Confirmed Path: $vp_path"
    app_dir="$vp_path"
}
```

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+                fi
+            ;;
+            n|N ) read -p "The path to your Visual Paradigm (APP FOLDER) is: " $1 ;;
+            * ) printf "Invalid input\n";;
+        esac
+    done
+    app_dir="$1"
+}
+
+function read_Plugin_Path(){
```
Comment (matheuslenke) [inline install.sh:40]: The same error happens with this function on Windows. Can you please change it?
```bash
function read_Plugin_Path(){
    plugin_path="$1"
    while true; do
        echo "Visual Paradigm Plugin Path: $plugin_path"
        read -p "Confirm (y/n)?: " choice
        case "$choice" in
            y|Y ) break;;
            n|N ) 
                echo "The path to your Visual Paradigm (PLUGIN FOLDER) is: "
                read -r plugin_path ;;
            * ) printf "Invalid input\n";;
        esac
    done
    plugin_dir="$plugin_path"
}
```

**Review by matheuslenke — CHANGES_REQUESTED** [review]
Hey @propilideno! I was taking a look at your code, and had some problems while installing it with Windows. I mainly use Unix systems, however, I need to make sure the experience is good even on Windows machines. 

The problem is when changing the path at functions `read_App_Path()` and `read_Plugin_Path()`. When answering with "n", it gives an error and does not allow the user to type a new path. I'm not sure what the reason is, but using another auxiliar variable seemed to fix it on what I tested.

Can you please fix it? If you can't, I will change it and approve your PR

**Comment by propilideno** [comment]
Hello @matheuslenke,
I appreciate your comments and i will review my code within 15 days.

Best Regards,
Lucas Almeida.

File: install.sh (hunk 1 above)
Comment (propilideno) [reply install.sh]: Fixed on [1a15e8c](https://github.com/OntoUML/ontouml-vp-plugin/pull/146/commits/1a15e8c0dbadab4c77a18e0df57a28ff273ec29b)

File: install.sh (hunk 2 above)
Comment (propilideno) [reply install.sh]: Fixed on [1a15e8c](https://github.com/OntoUML/ontouml-vp-plugin/pull/146/commits/1a15e8c0dbadab4c77a18e0df57a28ff273ec29b)

File: README.md
Comment (propilideno) [inline README.md:1]: Fixed on [89196a9](https://github.com/OntoUML/ontouml-vp-plugin/pull/146/commits/89196a9b9929b50440393c571444a866f05dcdc0)

**Review by propilideno — COMMENTED** [review]
Done.

**Comment by propilideno** [comment]
@matheuslenke

**Comment by matheuslenke** [comment]
Tested and everything seems fine to me. Thanks for your contribution, really nice the file picker on Windows. Merging now

**Pull request merged**